cmd/metal-api/internal/metal/size.go

```diff
@@ -84,6 +87,31 @@ func (c *Constraint) Matches(hw MachineHardware) (ConstraintMatchingLog, bool) {
 	case StorageConstraint:
 		res = hw.DiskCapacity() >= c.Min && hw.DiskCapacity() <= c.Max
 		cml.Log = fmt.Sprintf(logentryFmt, hw.DiskCapacity(), hw.DiskCapacity())
+	case GPUConstraint:
+		existing := make(map[string]uint8)
+		for _, gpu := range hw.MetalGPUs {
+			_, ok := existing[gpu.Model]
+			if !ok {
+				existing[gpu.Model] = 1
+			} else {
+				existing[gpu.Model]++
+			}
+		}
```

Comment:
i think the code

~~~go
existing[gpu.Model] = existing[gpu.Model] +1 
~~~

should also work.

Reply:
But why ?

Reply:
less verbose, but it's ok do to it this way.